clang/lib/Sema/SemaExpr.cpp

```diff
@@ -11247,6 +11247,12 @@ static void DiagnoseBadShiftValues(Sema& S, ExprResult &LHS, ExprResult &RHS,
   if (S.getLangOpts().OpenCL)
     return;
 
+  if (Opc == BO_Shr &&
+      LHS.get()->IgnoreParenImpCasts()->getType()->isBooleanType()) {
+    S.Diag(Loc, diag::warn_shift_bool) << LHS.get()->getSourceRange();
+    return;
```

Comment:
I don't think we want to return here, don't we want the rest of the checking? e.g.,

`true >> -1` should get both the shift bool warning and the negative shift warning?

Reply:
@AaronBallman, thanks for the feedback. I've removed the return to proceed with other diagnostics